filament/src/MaterialParser.cpp

```diff
@@ -181,7 +197,21 @@ uint32_t MaterialParser::computeCrc32() const noexcept {
 
     std::vector<uint32_t> crc32Table;
     utils::hash::crc32GenerateTable(crc32Table);
-    return utils::hash::crc32Update(0, payload, originalSize, crc32Table);
+    uint32_t crc32 = utils::hash::crc32Update(0, payload, originalSize, crc32Table);
+    mCrc32.store(crc32, std::memory_order_relaxed);
+    return crc32;
+}
+
+std::optional<uint32_t> MaterialParser::getPrecomputedCrc32() const noexcept {
+    std::optional<uint32_t> cachedCrc32 = mCrc32.load(std::memory_order_relaxed);
+    if (cachedCrc32) {
+        return cachedCrc32;
+    }
```

Comment:
same here, the line below returns a uint32_t, this should use the same pattern, no need to create a std::optional explicitely